flink-core/src/main/java/org/apache/flink/api/common/operators/GenericDataSinkBase.java

```diff
@@ -176,6 +182,31 @@ public void setLocalOrder(Ordering localOrder) {
 		this.localOrdering = localOrder;
 	}
 
+	public Ordering getPartitionOrdering() {
```

Comment:
`partitionByRange` with `DataDistribution` is added in `DataSet`, so we could `partitionByRange` with `DataDistribution` at any point of Flink DAG, these interface/logic is duplicated.